renv/utils.py

```diff
@@ -84,3 +89,31 @@ def create_symlink(src, dst, subfolders=[]):
                 Warning("Cannot create symlink for: " + dst_folder)
             os.symlink(src_folder, dst_folder)
 
+
+def system_r_call(rcmd_type, context):
+    """
+    Call the current R with system calls in order to obtain specific types
+    of information.
+    :param rcmd_type:  A string that designates the R command to use in the system call
+    :param context:  The context variable used in the R command.
+    :return:  Returns the stdout and stderr from the system call.
+    """
+
+    if rcmd_type == "major":
+        Rcmd = "%s -e \'R.version$major\'" % context.abs_R_script
+    elif rcmd_type == "minor":
+        Rcmd = "%s -e \'R.version$minor\'" % context.abs_R_script
+    elif rcmd_type == "base":
+        Rcmd = "%s -e \'base::cat(rownames(installed.packages(priority=\"base\")))\'" % context.abs_R_script
+    elif rcmd_type == "recommended":
+        Rcmd = "%s -e \'base::cat(rownames(installed.packages(priority=\"recommended\")))\'" % context.abs_R_script
+
+    recommended_pkgs = sp.Popen([Rcmd], stderr=sp.PIPE, stdout=sp.PIPE, shell=True, encoding='utf-8')
+
+    try:
+        stdout, stderr = recommended_pkgs.communicate(timeout=15)
+    except TimeoutExpired:
```

Comment:
Is this the only exception we might hit?

Reply:
There certainly could be others.  This particular exception handling block was just the most obvious one.  For now I'll create an issue to help resolve this in future PRs.

Reply:
#49